pkg/flow/flow_http.go

```diff
@@ -91,3 +94,38 @@ func (f *Flow) configBytes(w io.Writer, debugInfo bool) (n int64, err error) {
 
 	return file.WriteTo(w)
 }
+
+// ComponentJSON returns the json representation of the flow component.
+func (c *Flow) ComponentJSON(w io.Writer, ci *river.ComponentField) (int, error) {
+	var foundComponent *controller.ComponentNode
+	for _, c := range c.loader.Components() {
+		if c.ID().String() == ci.ID {
+			foundComponent = c
+			break
+		}
+	}
+	if foundComponent == nil {
+		return 0, fmt.Errorf("unable to find component named %s", ci.ID)
+	}
+	field, err := river.ConvertComponentToJSON(
+		foundComponent.ID(),
+		foundComponent.Arguments(),
+		foundComponent.Exports(),
+		foundComponent.DebugInfo(),
+		ci.References,
+		ci.ReferencedBy,
+		&river.Health{
+			State:       ci.Health.State,
+			Message:     ci.Health.Message,
+			UpdatedTime: ci.Health.UpdatedTime,
+		},
+		"")
+	if err != nil {
+		return 0, err
+	}
+	bb, err := json.MarshalIndent(field, "", "    ")
+	if err != nil {
+		return 0, err
+	}
```

Comment:
nit: don't emit pretty-printed JSON since it'll really blow up the output. Let tooling format the JSON if we need to look at it as humans.

Reply:
With GZIP enabled it added 10% size to pretty print. From a debugging standpoint, it's nice to not have to pretty print but if we think that's important I can remove it.